Scripts/Sorcery_Functions.py

```diff
@@ -0,0 +1,99 @@
+        print("Total Sorceries: %d" % Sorcery.sorceryCount)
+
+    def displayMagicAttributes(self):
+        print("Name: " + self.name + '\n' + "MP Cost: %d" % self.MPCost)
+        print("Element: " + self.element)
+
+    def getName(self):
+        return self.name
+
+    def getForbiddenStatus(self):
+        return self.forbidden
+
+# Creates some basic spells and returns a
+# list of them
+def createSpells(forbiddenSpellFlag, hippoFlag):
+    spellBook = []
+    fireBall = Sorcery("Fireball", 5, "Fire", False)
+    iceStorm = Sorcery("Ice Storm", 20, "Frost", False)
+    shadowDagger = Sorcery("Shadow Dagger", 7, "Dark", False)
+    arcaneShield = Sorcery("Arcane Shield", 12, "Arcane", False)
+
+    spellBook.append(fireBall)
+    spellBook.append(iceStorm)
+    spellBook.append(shadowDagger)
+    spellBook.append(arcaneShield)
+
+    if (forbiddenSpellFlag):
+        chaosFireball = Sorcery("Chaos Fireball", 20, "Fire", True)
+        raiseUndead = Sorcery("Raise Undead", 15, "Unholy", True)
+        shadowLance = Sorcery("Shadow Lance", 30, "Dark", True)
+        spellBook.append(chaosFireball)
+        spellBook.append(raiseUndead)
+        spellBook.append(shadowLance)
+
+    if (hippoFlag):
+        summonHippo = Sorcery("Summon Hippopotamus", 50, "Arcane", True)
+        spellBook.append(summonHippo)
+
+    return spellBook
+
+# The printSpellBook function takes in a spell book and
+# will print a list of the spells in the spell book. It
+# will not print the list if the size of the spell book
+# is 0. Additionally, the type of output printed will vary
+# depending on whether the spell book contains forbidden
+# spells
+def printSpellBook(spellBook):
+    hippoSpellName = "Summon Hippopotamus"
+    hippoFlag = False
+    forbiddenSpellFlag = False
+    hippoSpellLocation = -1
+
+    if (len(spellBook) <= 0):
+        print("The spell book is currently empty;")
+        print("Please add some spells if you are wanting to print them.")
+        return
+
+    print("Checking to see if all spells in the spell book are valid...")
+    for i in range(len(spellBook)):
+        if ((spellBook[i].getName() == hippoSpellName) and spellBook[i].getForbiddenStatus()):
+            hippoFlag = True
+            hippoSpellLocation = i
+            break
+        elif(spellBook[i].getForbiddenStatus()):
+            forbiddenSpellFlag = True
+
+    spellBook[0].displayCount()
+    if (hippoFlag):
+        print("The hippopotamus spell is too powerful. It ")
+        print("should not be added to the spellBook! ")
+        print("Here's some info on it anyway: ")
+        spellBook[hippoSpellLocation].displayMagicAttributes()
+
+    elif(forbiddenSpellFlag):
+        print("Caution: At least one forbidden spell was found.")
+        print("Here is a complete list of the forbidden spells")
+        print("found in the spell book:")
+        for i in range(len(spellBook)):
+            if (spellBook[i].getForbiddenStatus()):
```

Comment:
Something to make note of, you can write this for loop like this:

` for spell in spellbook:`
`    if spell.getForbidenStatus():`
`      spell.displayMagicAttributes()`

Couldn't format it cleanly, but give it a try.  I can break down why this works if you want.